packages/expo/cli/start/server/webpack/WebpackBundlerDevServer.ts

```diff
@@ -0,0 +1,351 @@
+import { createSymbolicateMiddleware } from '@expo/dev-server/build/webpack/symbolicateMiddleware';
+import chalk from 'chalk';
+import fs from 'fs';
+import http from 'http';
+import * as path from 'path';
+import resolveFrom from 'resolve-from';
+import type webpack from 'webpack';
+import type WebpackDevServer from 'webpack-dev-server';
+
+import * as Log from '../../../log';
+import { WEB_HOST } from '../../../utils/env';
+import { CommandError } from '../../../utils/errors';
+import { getIpAddress } from '../../../utils/ip';
+import { choosePortAsync } from '../../../utils/port';
+import { ensureDotExpoProjectDirectoryInitialized } from '../../project/dotExpo';
+import { BundlerDevServer, BundlerStartOptions, DevServerInstance } from '../BundlerDevServer';
+import { UrlCreator } from '../UrlCreator';
+import {
+  importExpoWebpackConfigFromProject,
+  importWebpackDevServerFromProject,
+  importWebpackFromProject,
+} from './resolveFromProject';
+import { ensureEnvironmentSupportsSSLAsync } from './ssl';
+
+type AnyCompiler = webpack.Compiler | webpack.MultiCompiler;
+
+export type WebpackConfiguration = webpack.Configuration & {
+  devServer?: { before?: Function };
+};
+
+function assertIsWebpackDevServer(value: any): asserts value is WebpackDevServer {
+  if (!value?.sockWrite) {
+    throw new CommandError(
+      'WEBPACK',
+      value ? 'Expected Webpack dev server, found: ' + value : 'Webpack dev server not started yet.'
```

Comment:
If `value` is an object, we'll get a useless `[object Object]` message. May want to try to print out value.constructor?.name if it is set.